src/main/java/sp/sd/nexusartifactuploader/steps/NexusArtifactUploaderStep.java

```diff
@@ -0,0 +1,333 @@
+package sp.sd.nexusartifactuploader.steps;
+
+/**
+ * Created by suresh on 5/19/2016.
+ */
+
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardUsernameCredentials;
+import com.cloudbees.plugins.credentials.common.StandardUsernameListBoxModel;
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import com.cloudbees.plugins.credentials.domains.DomainRequirement;
+import com.google.common.base.Strings;
+import hudson.*;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+
+import hudson.model.Item;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.remoting.VirtualChannel;
+import hudson.security.ACL;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import hudson.util.Secret;
+import jenkins.MasterToSlaveFileCallable;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.jenkinsci.remoting.RoleChecker;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+import sp.sd.nexusartifactuploader.Utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+
+public final class NexusArtifactUploaderStep extends AbstractStepImpl {
```

Comment:
@jglick
Yes both do same functionality, my point of implementation both step and builder is the declaration part different.
Builder uses the UI to configure in Build Steps 
But where as Step declaration is kind of function call for example 
step([$class: 'FunkyNotificationBuilder', artifact: 'myapp.exe']) 
So for user friendly type of declaration I have considered to implement the Step as well.

If this is not the practice followed by the Jenkins community means I will remove the Step.
Let me know I will plan for next release.